app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -182,6 +182,28 @@ def self.find_with_expiring_pools(_key, _operator, days_from_now)
           end
         end
 
+	def self.of_eos_index(release)
+	 version, arch = case release
+			   when 'RHEL7'
+			     ['7', 'x86_64']
+			   when 'RHEL7 (POWER9)'
+			     ['7', 'ppc64le']
+			   when 'RHEL7 (ARM)'
+			     ['7', 'aarch64']
+			   when 'RHEL7 (System z (Structure A))'
+			     ['7', 's390x']
+			   when 'RHEL9', 'RHEL8', 'RHEL6', 'RHEL5'
+			     /RHEL(?<version>\d+)/ =~ release
+			     [version]
+			   else
+			     fail("Not supported EOS index: #{release}")
+			   end
+
+	 query = joins(:operatingsystem).where("operatingsystems.name = ? and operatingsystems.major = ?", 'RedHat', version)
+	 query = query.joins(:architecture).where("architectures.name = ?", arch) if arch.present?
```

Comment:
It's clear now that operatingsystem `name == 'RedHat'` is not a reliable indicator of whether a system is RHEL. See https://github.com/Katello/katello/pull/10738, which is changing the RHEL lifecycle status not to rely on operatingsystem anymore. 

Once the approach on https://github.com/Katello/katello/pull/10738 is agreed upon and merged, this will need some refactoring.